core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -128,6 +131,15 @@ private void validateInputFields(Schema inputSchema) {
     } else if ((inputSchema.getField(conf.getFieldValue())) == null) {
       throw new IllegalArgumentException(
         String.format("Fieldvalue '%s' does not exist in input schema %s", conf.getFieldValue(), inputSchema));
+    } else if (!inputSchema.getField(conf.getKeyField()).getSchema().getType().equals(Schema.Type.STRING)) {
+      throw new IllegalArgumentException(
+        String.format("Keyfield '%s' must be of type String", conf.getKeyField()));
+    } else if (!inputSchema.getField(conf.getFieldName()).getSchema().getType().equals(Schema.Type.STRING)) {
+      throw new IllegalArgumentException(
+        String.format("Fieldname '%s' must be of type String", conf.getFieldName()));
+    } else if (!inputSchema.getField(conf.getFieldValue()).getSchema().getType().equals(Schema.Type.STRING)) {
```

Comment:
right but I see in the design that types are supposed to be converted to string, which means you could get an int or a long or something else here. Or is that not the case?